apph-front/src/static/components/EditProfile.tsx

```diff
@@ -0,0 +1,174 @@
+  const [login, setLogin] = useState('');
+  const [password, setPassword] = useState('');
+  const [passwordConfirmation, setPasswordConfirmation] = useState('');
+  const [alertMessage, setAlertMessage] = useState<string>();
+  const [errorOccured, setErrorOccured] = useState(false);
+  const [dialogOpen, setDialogOpen] = useState<boolean>(false);
+  const classes = makeCardStyles();
+
+  useEffect(() => {
+    (async () => {
+      await UserService.getUser(
+        (user: string) => {
+          const userConverted: IUser = JSON.parse(user);
+          setUser((user) => ({ ...userConverted }));
+          setFirstName(userConverted.firstname);
+          setLastName(userConverted.lastname);
+          setLogin(userConverted.login);
+        },
+        (errorMessage: string) => {
+          setErrorOccured(true);
+          setAlertMessage(errorMessage);
+        }
+      );
+    })();
+  }, []);
+
+  const editedFields = (): IUserRequest => {
+    return {
+      ...(firstName != user.firstname && { firstName }),
+      ...(lastName != user.lastname && { lastName }),
+      ...(login != user.login && { email: login }),
+      ...(password && { password })
+    };
+  };
+
+  const displayAlert = (message: string) => {
+    setAlertMessage(message);
+    document.getElementById('alert')?.scrollIntoView();
+  };
+
+  const editUser = async () =>
+    UserService.editUser(
+      editedFields(),
+      (newToken: string) => {
+        if (editedFields().email) {
+          AuthService.logout();
+          navigate(0);
+        } else {
+          AuthService.updateUserCookie(newToken);
+          setErrorOccured(false);
+          displayAlert('Le profil a bien été modifié.');
+          navigate('/me');
+        }
+      },
+      (errorMessage: string) => {
+        setErrorOccured(true);
+        displayAlert(errorMessage);
+      }
+    );
+
+  const handleSubmit = (event: FormEvent) => {
+    event.preventDefault();
+    if (password != passwordConfirmation) return;
+    if (editedFields().email) {
+      setDialogOpen(true);
+    } else {
+      editUser();
+    }
+  };
+
+  const resetChanges = () => {
+    setFirstName(user.firstname);
+    setLastName(user.lastname);
+    setLogin(user.login);
+  };
+
+  return (
+    <Card className={classes.cardStyle}>
+      <CardHeader title="Modifier le profil" />
+      <CardContent>
+        <Box component="form" onSubmit={handleSubmit}>
+          <Stack spacing={2}>
+            <TextField
+              required
+              label="Nom"
+              value={lastName}
+              onChange={(e) => setLastName(e.target.value)}
+            />
+            <TextField
+              required
+              label="Prénom"
+              value={firstName}
+              onChange={(e) => setFirstName(e.target.value)}
+            />
+            <TextField
+              required
+              label="Login"
+              value={login}
+              onChange={(e) => setLogin(e.target.value)}
```

Comment:
```suggestion
              onChange={(e) => setLogin(e.target.value)}
              type="email"              
```

rajoute ça pour verifier que l'email est valide